src/NLog/LoggerImpl.cs

```diff
@@ -167,12 +167,12 @@ internal static int SkipToUserStackFrameLegacy(StackFrame[] stackFrames, int fir
                 if (SkipAssembly(stackFrame))
                     continue;
 
-                if (stackFrame.GetMethod().Name == "MoveNext")
+                if (stackFrame.GetMethod()?.Name == "MoveNext")
                 {
                     if (stackFrames.Length > i)
                     {
                         var nextStackFrame = stackFrames[i + 1];
-                        var declaringType = nextStackFrame.GetMethod().DeclaringType;
+                        var declaringType = nextStackFrame.GetMethod()?.DeclaringType;
```

Comment:
.GetMethod() cannot return null now, correct?

Reply:
Not sure I understand. GetMethod() can return null, so free navigation can be dangerous. See the `stackFrames[i + 1]`.

Reply:
yes, missed that, thanks!